front/src/Components/Chat/UsersList.svelte

```diff
@@ -0,0 +1,155 @@
+<script lang="ts">
+    import type { Teleport, TeleportStore, User, UserList, UsersStore } from "../../Xmpp/MucRoom";
+    import LL from "../../i18n/i18n-svelte";
+    import {
+        walkAutomaticallyStore,
+        canShare,
+        getLink,
+        copyLink,
+        shareLink,
+        updateInputFieldValue,
+    } from "../../Stores/GuestMenuStore";
+    import { USER_STATUS_DISCONNECTED } from "../../Xmpp/MucRoom";
+    import { createEventDispatcher, onMount } from "svelte";
+    import { searchValue } from "../../Stores/Utils/SearchStore";
+    import { localUserStore } from "../../Connexion/LocalUserStore";
+
+    export let usersListStore: UsersStore;
+    let usersList: UserList = new Map<string, User>();
+
+    export let teleportStore: TeleportStore;
+    let teleport: Teleport = { state: false, to: null };
+
+    const dispatch = createEventDispatcher();
+
+    function goTo(type: string, roomId: string, uuid: string) {
+        dispatch("goTo", { type, roomId, uuid });
+    }
+
+    onMount(() => {
+        usersListStore.subscribe((value: UserList) => {
+            usersList = value;
+        });
+        teleportStore.subscribe((value: Teleport) => {
+            teleport = value;
+        });
+    });
+
+    //recreate user list
+    searchValue.subscribe((value: string | null) => {
+        usersList = new Map<string, User>();
+        usersListStore.subscribe((users: UserList) => {
```

Comment:
For each store you subscribe to, there should be a matching unsubscriber (otherwise, you are leaking memory)

But even better, with Svelte, you usually don't need to subscribe to a store in the Svelte component.
Instead, use the "$[store_name]" notation to get the reactive value of the store.

So you can use $usersListStore and $teleportStore.

For searchValue, you will need to unsubscribe during the unmount of the component.